tests/integration/test_storage_postgresql/test.py

```diff
@@ -625,6 +629,48 @@ def test_uuid(started_cluster):
     assert result.strip() == "Nullable(UUID)"
 
 
+def test_auto_close_connection(started_cluster):
+    conn = get_postgres_conn(
+        started_cluster.postgres_ip, started_cluster.postgres_port, database=False
+    )
+    cursor = conn.cursor()
+    database_name = "auto_close_connection_test"
+
+    cursor.execute(f"DROP DATABASE IF EXISTS {database_name}")
+    cursor.execute(f"CREATE DATABASE {database_name}")
+    conn = get_postgres_conn(
+        started_cluster.postgres_ip,
+        started_cluster.postgres_port,
+        database=True,
+        database_name=database_name,
+    )
+    cursor = conn.cursor()
+    cursor.execute("CREATE TABLE test_table (key integer, value integer)")
+
+    node2.query(
+        f"""
+        CREATE TABLE test.test_table (key UInt32, value UInt32)
+        ENGINE = PostgreSQL(postgres1, database='{database_name}', table='test_table')
+    """
+    )
+
+    result = node2.query(
+        "INSERT INTO test.test_table SELECT number, number FROM numbers(1000)",
+        user="default",
+    )
+
+    result = node2.query("SELECT * FROM test.test_table LIMIT 100", user="default")
+
+    count = int(
+        node2.query(
+            f"SELECT numbackends FROM postgresql(postgres1, database='{database_name}', table='pg_stat_database') WHERE datname = '{database_name}'"
+        )
+    )
+
+    # Connection from python
+    assert count == 1
```

Comment:
I guess nobody runs the integration test in parallel. Otherwise, this check can fail.